Nodejs/Product/Nodejs/Repl/NpmReplCommand.cs

```diff
@@ -59,38 +59,12 @@ class NpmReplCommand : IReplCommand {
             npmArguments = string.Format(" {0} ", npmArguments);
 
             var solution = Package.GetGlobalService(typeof(SVsSolution)) as IVsSolution;
-            IEnumerable<IVsProject> loadedProjects = solution.EnumerateLoadedProjects(onlyNodeProjects: true);
+            var loadedProjects = solution.EnumerateLoadedProjects(onlyNodeProjects: false);
 
-            var projectNameToDirectoryDictionary = new Dictionary<string, Tuple<string, IVsHierarchy>>(StringComparer.OrdinalIgnoreCase);
-            foreach (IVsProject project in loadedProjects) {
-                var hierarchy = (IVsHierarchy)project;
-                object extObject;
-
-                var projectResult = hierarchy.GetProperty(VSConstants.VSITEMID_ROOT, (int)__VSHPROPID.VSHPROPID_ExtObject, out extObject);
-                if (!ErrorHandler.Succeeded(projectResult)) {
-                    continue;
-                }
-
-                EnvDTE.Project dteProject = extObject as EnvDTE.Project;
-                if (dteProject == null) {
-                    continue;
-                }
-
-                EnvDTE.Properties properties = dteProject.Properties;
-                if (dteProject.Properties == null) {
-                    continue;
-                }
-
-                string projectName = dteProject.Name;
-                EnvDTE.Property projectHome = properties.Item("ProjectHome");
-                if (projectHome == null || projectName == null) {
-                    continue;
-                }
-
-                var projectDirectory = projectHome.Value as string;
-                if (projectDirectory != null) {
-                    projectNameToDirectoryDictionary.Add(projectName, Tuple.Create(projectDirectory, hierarchy));
-                }
+            // Prefer Node projects, but fall back to a non-node project if no Node projects are loaded.
+            var projectNameToDirectoryDictionary = GetProjectInfo(solution.EnumerateLoadedProjects(onlyNodeProjects: true));
+            if (!projectNameToDirectoryDictionary.Any()) {
+                projectNameToDirectoryDictionary = GetProjectInfo(solution.EnumerateLoadedProjects(onlyNodeProjects: false));
             }
```

Comment:
This part may be attempting to be too clever.

In a multi-project solution, it may be better to require users to explicitly state which project they want to run NPM for. Currently, if there is any Node project in a solution, you can't run .npm for any of the other projects in that solution.